Automattic/wp-calypso
# Jetpack Pricing Cart Integration - Handle CTA Label

#### Proposed Changes

- Make changes to masterbar-cart-button to have the state in redux instead of inside the component so that we can control the state from other components - create cartOpen state in ui redux
- Changes to show the correct button label with icon when an item is added to cart
- Dispatch state update to open the cart on click of View Cart CTA on any items

#### Testing Instructions

* click on Jetpack Cloud live link below
    * Goto `/pricing`
* or boot up this PR 
    * Run `git fetch && git checkout feature/cloud-cart-feature-flag
    * Run `yarn start-jetpack-cloud`
    * Goto http://jetpack.cloud.localhost:3000/pricing
* Switch to Bundle section and confirm the button Label on the card is the same as before and confirm that clicking on it takes you to checkout page
* Switch to the Product section and confirm the following
    * All the product cards (except the one already purcahsed) CTA label is **Add to Cart** instead of **Get**
    * Clicking on the CTA adds the item to the cart in header and button label for the item changes to **View Cart**
    * Make sure the **View Cart** CTA matches the design
    * Click on **View Cart** open the cart and make sure the item is there in the cart
 
#### Screenshot

##### Before
![Screenshot 2022-12-12 at 5 58 29 PM](https://user-images.githubusercontent.com/2027003/207045466-ce881498-3244-4f38-9a47-1ea0e85be528.png)

##### After
![Screenshot 2022-12-12 at 5 57 44 PM](https://user-images.githubusercontent.com/2027003/207045451-b400f849-40ac-4add-b2ca-ed12c3fa1dcd.png)


#### Pre-merge Checklist





Related to #
